packages/flutter/test/foundation/leak_tracking.dart

```diff
@@ -5,8 +5,115 @@
 import 'package:flutter/foundation.dart';
 import 'package:flutter_test/flutter_test.dart';
 import 'package:leak_tracker/leak_tracker.dart';
+import 'package:meta/meta.dart';
 
-typedef LeaksObtainer = void Function(Leaks foundLeaks);
+/// Set of objects, that does not hold the objects from garbafge collection.
+///
+/// The objects are referenced by hash codes and can duplicate with low probability.
+@visibleForTesting
+class WeakSet {
+  final Set<String> _objectCodes = <String>{};
+
+  String _toCode(int hashCode, String type) => '$type-$hashCode';
+
+  void add(Object object) {
+    _objectCodes.add(_toCode(identityHashCode(object), object.runtimeType.toString()));
+  }
+
+  void addByCode(int hashCode, String type) {
+    _objectCodes.add(_toCode(hashCode, type));
+  }
+
+  bool contains(int hashCode, String type) {
+    final bool result = _objectCodes.contains(_toCode(hashCode, type));
+    return result;
+  }
+}
+
+@visibleForTesting
+class TestWidgetsConfig {
+  final WeakSet heldObjects = WeakSet();
+}
+
+/// Extension for [WidgetTester] to configure held objects.
+extension LeakTrackerConfig on WidgetTester {
+  static final Expando<TestWidgetsConfig> _configExpando = Expando<TestWidgetsConfig>();
+
+  T addHeldObject<T  extends Object>(T object){
+    _config.heldObjects.add(object);
+    return object;
+  }
+
+  bool isHeld(Object object){
+    return _config.heldObjects.contains(
+      identityHashCode(object),
+      object.runtimeType.toString(),
+    );
+  }
+
+  TestWidgetsConfig get _config {
+    if (_configExpando[this] == null) {
+      _configExpando[this] = TestWidgetsConfig();
+    }
+    return _configExpando[this]!;
+  }
+}
+
+/// Wrapper for [testWidgets] with leak tracking.
+///
+/// Tracking for non-GCed objects is temporarily disabled.
```

Comment:
Do we have an issue on file that details why this is? Would be nice to link it from here.